gcloud/orb.yaml

```diff
@@ -252,6 +296,102 @@ jobs:
+        type: string
+      k8s_yaml:
+        description: >
+          The k8s YAML file for your deployment(s).
+        type: string
+      project:
+        type: string
+      prune_selector:
+        description: >
+          A k8s label selector to identify the k8s deployments that are
+          relevant to this deploy step, and prune those that aren't present in
+          the k8s YAML. Default is not to prune.
+        type: string
+        default: ""
+      version_selector:
+        description: >
+          A k8s label selector to identify which deployment should be recorded
+          as the previous deployment version for release notifications. If not
+          specified, the deployment name is used.
+        type: string
+        default: ""
+      zone:
+        type: string
+    steps:
+      - auth:
+          creds: <<parameters.creds>>
+          project: <<parameters.project>>
+      - configure-gke:
+          cluster: <<parameters.cluster>>
+          zone: <<parameters.zone>>
+      - run: mkdir -p /metadata/prev_version/<<parameters.deployment>>
+      - when:
+          condition: <<parameters.version_selector>>
+          steps:
+            - run:
+                name: store previous version
+                command: |
+                  # Get the deployment image from the currently deployed k8s
+                  # manifest, parse out the tag as the last (3rd) chunk of img, and
+                  # further split the tag to retrieve the rightmost chunk: the commit
+                  # hash.
+                  kubectl get deployments -l <<parameters.version_selector>> -oyaml | awk '/image: / {split($0,img,":"); len=split(img[3],tag,"-"); print tag[len]}' | tail -n1 > /metadata/prev_version/<<parameters.deployment>>/<<parameters.zone>>
+      - unless:
+          condition: <<parameters.version_selector>>
+          steps:
+            - run:
+                name: store previous version
+                command: |
+                  kubectl get deployments <<parameters.deployment>> -oyaml | awk '/image: / {split($0,img,":"); len=split(img[3],tag,"-"); print tag[len]}' | tail -n1 > /metadata/prev_version/<<parameters.deployment>>/<<parameters.zone>>
+      - when:
+          condition: <<parameters.prune_selector>>
+          steps:
+            - run:
+                name: deploy to k8s cluster
+                command: |
+                  kubectl apply -f <<parameters.k8s_yaml>> --prune -l <<parameters.prune_selector>>
+      - unless:
+          condition: <<parameters.prune_selector>>
+          steps:
+            - run:
+                name: deploy to k8s cluster
+                command: |
+                  kubectl apply -f <<parameters.k8s_yaml>>
+      - when:
+          condition: <<parameters.autoscaler_yaml>>
+          steps:
+            - update-autoscaler:
+                config: <<parameters.autoscaler_yaml>>
+                entry_name: <<parameters.deployment>>.yaml
```

Comment:
This feels like a bit of an "and also" situation, ie. a case where this job is doing more than one specific thing. Would it be reasonable to do this via `post-steps` or such instead, within the relevant app repos? Feels kind of similar to how you have the interpolation piece pulled out into pre-steps.